Add API reference for `recompress_chunk` (#966)
timescale/docs: api/recompress_chunk.md

# recompress_chunk() <tag type="community" content="Community" />
Recompresses a chunk that was previously compressed and decompressed.
```sql
recompress_chunk(
    chunk REGCLASS,
    if_not_compressed BOOLEAN = false
)
```

> You can also recompress chunks by [running the job associated with your
> compression policy][run-job]. `recompress_chunk` gives you more fine-grained
> control by allowing you to target a specific chunk.

Reviewer: ```suggestion
You can also recompress chunks by 
[running the job associated with your compression policy][run-job]. 
`recompress_chunk` gives you more fine-grained control by 
allowing you to target a specific chunk.
```

<highlight type="important">

> `recompress_chunk` is implemented as a SQL procedure and not a function. Call

Reviewer: ```suggestion
`recompress_chunk` is implemented as an SQL procedure and not a function. Call
```

> the procedure with `CALL` rather than using a `SELECT` statement.

Reviewer: ```suggestion
the procedure with `CALL`, do not use a `SELECT` statement.
```

Author: ```suggestion
the procedure with `CALL`. Don't use a `SELECT` statement.
```

</highlight>

<highlight type="note">

> `recompress_chunk` only works on previously compressed chunks. To compress a

Reviewer: ```suggestion
`recompress_chunk` only works on chunks that have previously been compressed. To compress a
```

chunk for the first time, use [`compress_chunk`](/compression/compress_chunk/).
</highlight>

## Required arguments

|Name|Type|Description|
|-|-|-|
|`chunk`|`REGCLASS`|The chunk to be recompressed. Must preface with the schema, `_timescaledb_internal`.|

## Optional arguments

|Name|Type|Description|
|-|-|-|
|`if_not_compressed`|`BOOLEAN`|If `true`, prints a notice instead of erroring if the chunk is already compressed. Defaults to `false`.|

## Sample usage
Recompress the chunk `timescaledb_internal._hyper_1_2_chunk`:
```sql
CALL recompress_chunk('_timescaledb_internal._hyper_1_2_chunk');
```

## Troubleshooting

> Since TimescaleDB 2.6.0, `recompress_chunk` is implemented as a procedure.

Reviewer: ```suggestion
In TimescaleDB 2.6.0 and above, `recompress_chunk` is implemented as a procedure.
```

Previously, it was implemented as a function. If you are upgrading to

> TimescaleDB 2.6.0 or later, older code that uses `recompress_chunk` as a

Reviewer: ```suggestion
TimescaleDB 2.6.0 or above, the`recompress_chunk` 
```

> function may cause an error. For example, trying to run `SELECT

Reviewer: ```suggestion
function could cause an error. For example, trying to run `SELECT
```

recompress_chunk(i.show_chunks, true) FROM...` gives the following error:
```sql
ERROR:  recompress_chunk(regclass, boolean) is a procedure
```

> To fix the error, use `CALL` instead of `SELECT`. You may need to write a

Reviewer: ```suggestion
To fix the error, use `CALL` instead of `SELECT`. You might also need to write a
```

# recompress_chunk() <tag type="community" content="Community" />

> Recompresses a chunk that was previously compressed and decompressed.

Reviewer 2: ```suggestion
Recompresses a chunk that is compressed and had INSERTs after compression.
```

```sql
recompress_chunk(
    chunk REGCLASS,
    if_not_compressed BOOLEAN = false
)
```

You can also recompress chunks by 
[running the job associated with your compression policy][run-job]. 
`recompress_chunk` gives you more fine-grained control by 
allowing you to target a specific chunk.

<highlight type="important">
`recompress_chunk` is implemented as an SQL procedure and not a function. Call
the procedure with `CALL`. Don't use a `SELECT` statement.
</highlight>

<highlight type="note">
`recompress_chunk` only works on chunks that have previously been compressed. To compress a
chunk for the first time, use [`compress_chunk`](/compression/compress_chunk/).
</highlight>

## Required arguments

|Name|Type|Description|
|-|-|-|

> |`chunk`|`REGCLASS`|The chunk to be recompressed. Must preface with the schema, `_timescaledb_internal`.|

Reviewer 2: ```suggestion
|`chunk`|`REGCLASS`|The chunk to be recompressed. Must include the schema if not in `search_path`, e.g. `_timescaledb_internal`.|
```
